string shot
string mode
float32 angular_z
float32 angular_y
